packages/material-ui-styles/src/index.d.ts

```diff
@@ -200,16 +200,13 @@ declare module '@material-ui/styles/StylesProvider' {
 }
 
 declare module '@material-ui/styles/ThemeProvider' {
-  import { Theme } from '@material-ui/core';
-
-  const ThemeContext: React.Context<Theme>;
-
-  export interface ThemeProviderProps {
+  export interface ThemeProviderProps<Theme> {
```

Comment:
import { Theme as MuiTheme } from '@material-ui/core';

export interface ThemeProviderProps<Theme = MuiTheme> {

Reply:
I don't think we're using the MuiTheme as the default theme here.

Reply:
@eps1lon , I also think it should stay generic here, and only packages that consume this one, should be specific. But probably @VincentLanglet suggest this to stay "backward" compatible. However I'm not sure it this really breaks something.

Reply:
There shouldn't be a breaking change. If we pass in the mui theme then it would already infer that. This change makes the separation between core and styles more explicit.

Reply:
@VincentLanglet do you still think we need a default value for Theme parameters?

Reply:
I suggested this for backward compatibility, but that’s fine to me